questionpy/_ui.py

```diff
@@ -1,44 +1,79 @@
+from questionpy_common.environment import Package, PackageNamespaceAndShortName, get_qpy_environment
+from questionpy_sdk.constants import TEMPLATES_DIR
 
 if TYPE_CHECKING:
     from questionpy import Attempt, Question
 
 
-def _loader_for_package(package: Package) -> jinja2.BaseLoader | None:
-    pkg_name = f"{package.manifest.namespace}.{package.manifest.short_name}"
-    if not (importlib.resources.files(pkg_name) / "templates").is_dir():
-        # The package has no "templates" directory, which would cause PackageLoader to raise an unhelpful ValueError.
+class _TraversableTemplateLoader(jinja2.BaseLoader):
+    """In contrast to the :class:`jinja2.FileSystemLoader` this does not support the auto-reload feature of jinja2."""
+
+    def __init__(self, traversable: Traversable):
+        self.traversable = traversable
+
+    def get_source(self, environment: jinja2.Environment, template: str) -> tuple[str, str, Callable[[], bool] | None]:
+        source_path = self.traversable.joinpath(template)
+        try:
+            source = source_path.read_text("utf-8")
+        except FileNotFoundError as e:
+            raise jinja2.TemplateNotFound(template) from e
+        else:
+            return source, template, None
+
+
+def _get_loader(package: Package) -> jinja2.BaseLoader | None:
+    templates_folder = package.get_path(f"{TEMPLATES_DIR}/")
+    if not templates_folder.is_dir():
         return None
 
-    # TODO: This looks for templates in python/<namespace>/<short_name>/templates, we might want to support a different
-    #  directory, such as resources/templates.
-    return jinja2.PackageLoader(pkg_name)
+    if os.path.exists(str(templates_folder)):
+        return jinja2.FileSystemLoader(str(templates_folder))
+    return _TraversableTemplateLoader(templates_folder)
 
 
 def create_jinja2_environment(attempt: "Attempt", question: "Question") -> jinja2.Environment:
     """Creates a Jinja2 environment with sensible default configuration.
 
     - Library templates are accessible under the prefix ``qpy/``.
-    - Package templates are accessible under the prefix ``<namespace>.<short_name>/``.
+    - Package templates are accessible under the prefix ``@<namespace>.<short_name>/``.
+    - The prefix is optional when accessing templates of the current package.
     - The QPy environment, attempt, question and question type are available as globals.
     """
     qpy_env = get_qpy_environment()
 
     loader_mapping = {}
     for package in qpy_env.packages.values():
-        loader = _loader_for_package(package)
+        loader = _get_loader(package)
         if loader:
-            loader_mapping[f"{package.manifest.namespace}.{package.manifest.short_name}"] = loader
+            loader_mapping[f"@{package.manifest.namespace}.{package.manifest.short_name}"] = loader
 
     # Add a place for SDK-Templates, such as the one used by ComposedAttempt etc.
     loader_mapping["qpy"] = jinja2.PackageLoader(__package__)
```

Comment:
Dieser Präfix kollidiert mit (lokalen) Templates, deren Name/Pfad mit "qpy" beginnt. Vielleicht können wir hier auch ein "@" voranstellen? Und könnte der Schrägstrich mit in den Präfix aufgenommen werden?

```suggestion
    loader_mapping["@qpy/"] = jinja2.PackageLoader(__package__)
```

Reply:
`@qpy` würde wiederum mit Templates kollidieren, die diesen Namespace haben. Ich glaube auch, dass das `qpy` gar nicht mehr benötigt wird. Dazu frage ich gleich mal @MHajoha zur Sicherheit.

Reply:
`qpy/question.xhtml.j2` wird nicht mehr verwendet, seitdem wir die UI-Teile separat ausliefern.
`qpy/subquestion.xhtml.j2` wurde nur verwendet, als ich mit question composition experimentiert hatte.
Können eigentlich beide für's erste weg. Das Thema Composition kommt bestimmt irgendwann wieder, aber wie wir den Template-Präfix dafür nennen, können wir uns dann überlegen.